Formula/o/odin.rb

```diff
@@ -128,36 +156,34 @@ def install
     EOS
     system bin/"odin", "run", "miniaudio.odin", "-file"
 
-    if OS.mac?
-      (testpath/"raylib.odin").write <<~EOS
-        package main
-
-        import rl "vendor:raylib"
-
-        main :: proc() {
-          // raygui.
-          assert(!rl.GuiIsLocked())
-
-          // raylib.
-          num := rl.GetRandomValue(42, 1337)
-          assert(42 <= num && num <= 1337)
-        }
-      EOS
-      system bin/"odin", "run", "raylib.odin", "-file"
-      system bin/"odin", "run", "raylib.odin", "-file",
-        "-define:RAYLIB_SHARED=true", "-define:RAYGUI_SHARED=true"
-
-      (testpath/"glfw.odin").write <<~EOS
-        package main
-
-        import "core:fmt"
-        import "vendor:glfw"
-
-        main :: proc() {
-          fmt.println(glfw.GetVersion())
-        }
-      EOS
-      system bin/"odin", "run", "glfw.odin", "-file"
-    end
+    (testpath/"raylib.odin").write <<~EOS
+      package main
+
+      import rl "vendor:raylib"
+
+      main :: proc() {
+        // raygui.
+        assert(!rl.GuiIsLocked())
+
+        // raylib.
+        num := rl.GetRandomValue(42, 1337)
+        assert(42 <= num && num <= 1337)
+      }
+    EOS
+    system bin/"odin", "run", "raylib.odin", "-file"
+    # system bin/"odin", "run", "raylib.odin", "-file",
+    #   "-define:RAYLIB_SHARED=true", "-define:RAYGUI_SHARED=true"
```

Comment:
Why is this commented out?

Reply:
that is the code for dynamic linking for the raylib and raygui but since the path to vender libraries is not specified in env it is unable to link it

Reply:
If `odin` can handle shared libraries on macOS (as the original test clearly showed) but not Linux, this should be highlighted upstream. I suspect it's more a matter of getting the Linux build config right though.

Reply:
The problem is with installing raylib as a shared library on linux as when the shared library flag is used `odin` looking for the `.so` files in the current `directory/linux` instead of looking in the place where they exist as the linking is not done correctly

Reply:
If shared library linking from Odin's vendor directories isn't working on Linux, you should ask upstream what's going on, because it works for macOS (otherwise the flags you commented out wouldn't have been written into the test in the first place).

Reply:
I will have a look at this PR soon!

Reply:
Looks like you figured it out but just have some code style issues blocking the PR?

Reply:
no actually it is still failing while detecting the shared library on linux

Reply:
the issue is that the linker is looking the `.so` files in `current directory/linux/` but it has to look at a different location. I am unable to understand why that is happening I thought it worked but the only solution that I tried was to copy the files to the directory where it is currently looking in and that way it works

Reply:
@laytan could you please take a look why the linker is looking in the wrong directory because I have tired setting the LD_LIBRARY_SHARED path to point to the corrent location yet it is failing.

Reply:
Yeah so shared libraries are indeed kind of broken.

The reason it does work for macos is because brew rewrites `@rpath` to the absolute path of the library.

Currently, Odin wants you to copy the shared library into the `$(pwd)/linux/libraylib.so.500` but that is not a good thing and I will be bringing this up internally and seeing if we can unify this with all the other platforms and provide better errors.

We do want to have the user (or maybe Odin could do that automatically) copy over shared libraries into a known path though, if you think about shipping software, you can't rely on linking to a shared library that is at `/foo/whatever.so`, when you ship you want that to be relative to your executable so you can bundle up a release and have the shared libraries load on any system.

I think in the near future it makes sense to make changes so it works consistently the same everywhere, where the same is: expect the user to copy the shared libraries next to the executable. So `$executable/libraylib.so.500` and not `$(pwd)/linux/libraylib.so.500`. But we still need to work this all out and talk it over.

---

Because this is an upstream issue that also happens on our own releases, I think this should not be blocking this PR. Maybe skipping the test with a comment is the best to get this through.